src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -491,57 +491,159 @@ public void canCreateUpdateAndDeleteBookingRules() throws IOException, SQLExcept
     }
 
     @Test
-    void canCreateUpdateAndDeleteStopAreas() throws IOException, SQLException, InvalidNamespaceException {
+    void canCreateUpdateAndDeleteNonFlexPatternStop() throws IOException, SQLException, InvalidNamespaceException {
         // Store Table and Class values for use in test.
-        final Table stopAreasTable = Table.STOP_AREAS;
-        final Class<StopAreaDTO> stopAreaDTOClass = StopAreaDTO.class;
+        final Table patternStopTable = Table.PATTERN_STOP;
+        final Class<PatternStopDTO> patternStopDTOClass = PatternStopDTO.class;
 
         // create new object to be saved
-        final String areaId = "4153";
-        StopAreaDTO createdStopArea = createSimpleTestStopArea(areaId);
+        final String patternId = "pattern-id-1";
+        PatternStopDTO createdPatternStop = new PatternStopDTO(patternId, "stop-id-1", 0, 0);
+        JdbcTableWriter createTableWriter = createTestTableWriter(patternStopTable);
+        String output = createTableWriter.create(mapper.writeValueAsString(createdPatternStop), true);
+        createdPatternStop = mapper.readValue(output, PatternStopDTO.class);
+
+        // make sure saved data matches expected data
+        assertThat(createdPatternStop.pattern_id, equalTo(patternId));
+        assertThatSqlQueryYieldsRowCount(getColumnsForId(createdPatternStop.id, patternStopTable), 1);
+        // try to update record
+        final String updatedPatternStopId = "updated-pattern-id-2";
+        createdPatternStop.pattern_id = updatedPatternStopId;
+
+        // convert object to json and save it
+        JdbcTableWriter updateTableWriter = createTestTableWriter(patternStopTable);
+        String updateOutput = updateTableWriter.update(
+            createdPatternStop.id,
+            mapper.writeValueAsString(createdPatternStop),
+            true
+        );
+        LOG.info("update {} output:", patternStopTable.name);
+        LOG.info(updateOutput);
+
+        PatternStopDTO updatedPatternStopDTO = mapper.readValue(updateOutput, patternStopDTOClass);
+
+        // make sure saved data matches expected data
+        assertThat(updatedPatternStopDTO.pattern_id, equalTo(updatedPatternStopId));
+        // Verify that certain values are correctly set in the database.
+        ResultSet resultSet = getResultSetForId(updatedPatternStopDTO.id, patternStopTable);
+        while (resultSet.next()) {
+            assertResultValue(resultSet, "pattern_id", equalTo(createdPatternStop.pattern_id));
+        }
+        // try to delete record
+        JdbcTableWriter deleteTableWriter = createTestTableWriter(patternStopTable);
+        int deleteOutput = deleteTableWriter.delete(
+            createdPatternStop.id,
+            true
+        );
+        LOG.info("deleted {} records from {}", deleteOutput, patternStopTable.name);
```

Comment:
Extract a method for these statements that recur a few more times in this file.

Reply:
Ended up refactoring. Much cleaner now.